### PR TITLE
Mothur datatypes: fix metadata failure for large count tables

### PR DESCRIPTION
**Pull request by shiltemann**

### CODE REVIEW

File: lib/galaxy/datatypes/mothur.py [hunk 1]
```diff
@@ -270,7 +270,8 @@ def set_meta(self, dataset, overwrite=True, **kwd):
         dataset.metadata.column_names = self.column_names
         dataset.metadata.column_types = self.column_types
         dataset.metadata.comment_lines = self.comment_lines
-        dataset.metadata.data_lines -= self.comment_lines
+        if isinstance(dataset.metadata.datalines, int):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/mothur.py]: s/datalines/data_lines/ I suppose

File: lib/galaxy/datatypes/mothur.py [hunk 2]
```diff
@@ -735,7 +736,8 @@ def set_meta(self, dataset, overwrite=True, skip=1, max_data_lines=None, **kwd):
             dataset.metadata.groups = colnames[2:]
 
         dataset.metadata.comment_lines = 1
-        dataset.metadata.data_lines -= 1
+        if isinstance(dataset.metadata.datalines, int):
```
Comment (nsoranzo) [inline lib/galaxy/datatypes/mothur.py]: Same as above.

**Comment by shiltemann** [comment]
huh, how did I mess up copy pasting? :p friday afternoon I suppose.. thanks
@nsoranzo, will fix

**Comment by jmchilton** [comment]
Thanks @shiltemann. The failing API test case is unrelated and would have been fixed by #3668.

**Pull request merged**